app/controllers/concerns/api/users_controller.rb

```diff
@@ -43,7 +47,19 @@ def suspend
 
     def unpublish
       authorize(@user, :unpublish_all_articles?)
-      Moderator::UnpublishAllArticlesWorker.perform_async(params[:id].to_i)
+      target_articles = Article.published.where(user_id: params[:id].to_i)
+
+      unless target_articles.empty?
+        Moderator::UnpublishAllArticlesWorker.perform_async(params[:id].to_i)
+
+        payload = {
+          action: "api_user_unpublish",
+          target_user_id: params[:id].to_i,
+          target_article_ids: target_articles.ids
+        }
+        Audit::Logger.log(:admin_api, @user, payload)
```

Comment:
I added this check to avoid performing the unpublish action + creating a log for it if there are no articles to unpublish. But it's maybe worth keeping a log of the request itself even if no action was taken? Curious if anyone feels this way 🙂

Reply:
Hmm, I could go either way! It certainly would be helpful to know if someone is fishing around for articles to unpublished, but since we restrict these endpoints to admins I feel like the threat is low (and such activity can be monitored via data dog on Dev.to, or whatever monitoring a forem admin has hooked up

Reply:
I agree with what you mention. Leaning on the "more visibility over actions taken with these endpoints" makes sense. Done 👍🏼